internal/generator/vector/output/cloudwatch/cloudwatch.go

```diff
@@ -107,17 +109,22 @@ func sink(id string, o obs.OutputSpec, inputs []string, secrets observability.Se
 		Inputs:         vectorhelpers.MakeInputs(inputs...),
 		Region:         region,
 		GroupName:      groupName,
-		SecurityConfig: authConfig(o.Cloudwatch.Authentication, secrets),
+		SecurityConfig: authConfig(o.Name, o.Cloudwatch.Authentication, secrets, op),
 		EndpointConfig: endpointConfig(o.Cloudwatch),
 		RootMixin:      common.NewRootMixin("none"),
 	}
 }
 
-func authConfig(auth *obs.CloudwatchAuthentication, secrets observability.Secrets) Element {
+func authConfig(outputName string, auth *obs.CloudwatchAuthentication, secrets observability.Secrets, options Options) Element {
 	authConfig := NewAuth()
 	if auth != nil && auth.Type == obs.CloudwatchAuthTypeAccessKey {
 		authConfig.KeyID.Value = vectorhelpers.SecretFrom(&auth.AWSAccessKey.KeyId)
 		authConfig.KeySecret.Value = vectorhelpers.SecretFrom(&auth.AWSAccessKey.KeySecret)
+	} else if auth != nil && auth.Type == obs.CloudwatchAuthTypeIAMRole {
+		if forwarderName, ok := options[OptionForwarderName]; ok {
```

Comment:
Please look at utility methods on "options" .  I believe there is a function to make the retrieval and fetch/cast to a string cleaner